### PR TITLE
v3 - Add benefitgroup for a spesific location

### PR DESCRIPTION
**Pull request by anemne**
## Short Description
Benefit group is added to a spesific location, similar to bonus and salaryData. 

## Visual Overview (Image/Video)
Screenshot 2024-09-05 at 11 36 23
Screenshot 2024-09-05 at 11 37 35

## Checklist

Please ensure that you’ve completed the following checkpoints before submitting your pull request:

- [x] **Documentation**: Relevant documentation has been added or updated (if applicable).
- [x] **Testing**: Have you tested your changes thoroughly?
  - Please list the types of tests you've run (unit, integration, manual, etc.):

### CODE REVIEW

File: studio/schemas/objects/compensations/benefitsByLocation.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  },
+},
+],
+validation: (Rule) =>
+Rule.custom((benefitsByLocation) => {
+  return (
+    checkForDuplicateLocations(
+      benefitsByLocation as DocumentWithLocation[] | undefined,
+    ) ||
+    "Each location should be listed only once in the benefits list."
```
Comment (christinaroise) [inline studio/schemas/objects/compensations/benefitsByLocation.ts]:
```suggestion
    "Each location should be listed only once in the List of Benefits."
```

File: studio/utils/checkForDuplicateLocations.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+interface LocationReference {
```
Comment (christinaroise) [inline studio/utils/checkForDuplicateLocations.ts:1]: is this a new util or was it not added somewhere by @mathiazom

Comment (anemne) [reply studio/utils/checkForDuplicateLocations.ts:1]: Yes it is, i just needed it for the location validation

Comment (christinaroise) [reply studio/utils/checkForDuplicateLocations.ts:1]: ah!

File: studio/schemas/objects/compensations/benefitsByLocation.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            title: location
+              ? `Benefits group for ${location}`
+              : "No location selected",
+            subtitle: `Number of benefits: ${benefitsCount}`,
+          };
+        },
+      },
+    },
+  ],
+  validation: (Rule) =>
```
Comment (christinaroise) [inline studio/schemas/objects/compensations/benefitsByLocation.ts:52]: Not sure what we'll land on here, but can you apply the same approach once it's decided?
https://github.com/varianter/variant.no/pull/535/files#r1744997509

**Review by christinaroise — APPROVED** [review]

File: studio/utils/checkForDuplicateLocations.ts (hunk 2 above)
Comment (anemne) [reply studio/utils/checkForDuplicateLocations.ts:1]: Have you updated it in bonusesByLocation @mathias? I can do it if not ☺️

File: studio/schemas/objects/compensations/benefitsByLocation.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import {
+  checkForDuplicateLocations,
+  DocumentWithLocation,
+} from "studio/utils/checkForDuplicateLocations";
+
+export const benefitsByLocation = defineField({
+  name: "benefitsByLocation",
+  title: "Benefits by Location",
+  description:
+    "Enter the benefits offered at each office location. You can only have one location entry, but you can list multiple benefits for each office location.",
```
Comment (mathiazom) [inline studio/schemas/objects/compensations/benefitsByLocation.ts:14]:
```suggestion
    "Specify the benefits offered at each office location. A benefit is only associated with a single location",
```

File: studio/utils/checkForDuplicateLocations.ts (hunk 2 above)
Comment (mathiazom) [reply studio/utils/checkForDuplicateLocations.ts:1]: Yes, I have updated `bonusesByLocation` to use the common util in #535

**Review by mathiazom — APPROVED** [review]

**Pull request merged**